elytron/WFCORE-3947-ocsp.adoc

```diff
@@ -0,0 +1,67 @@
+= WFCORE-3947 OCSP support in Elytron SSL truststore
+:author:            Jan Kalina
+:email:             jkalina@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          elytron
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Elytron provides trustmanager usable for SSL contexts used by other parts of WildFly.
+It supports certificate revocations using certificate revocations list (CRL), but support of Online Certificate Status Protocol (OCSP) is required too.
+This is required especially for client certificates validation.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-715[EAP7-715]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFCORE-3947[WFCORE-3947]
+
+=== Dev Contacts
+
+* mailto:jkalina@redhat.com[Jan Kalina]
+
+=== QE Contacts
+
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly Elytron
+
+//=== Other Interested Projects
+
+== Requirements
```

Comment:
Could you please describe what actually will be configurable and how? E.g. shall we support to configure only one OCSP responder at one moment? Or shall we allow to have some backup one (or even a list of backup responders) in case that the primary is not available? What are timeouts in such case? Is the timeout configured per every such responder or per whole request?

Maybe I complicate things too much, but we should probably have this described and clarified here. I guess that simplest solution should be good enough?

Also, what are the defaults? Is there gonna be some default OCSP responder provided such as [1]?

What is the default reaction when OCSP response is UNKNOWN or is not available at all? Is this still part of this RFE?

[1] http://wiki.cacert.org/OcspResponder

Reply:
I has not find any backup OCSP requirement, so will not add it until it will be requested.
As it is not possible to specify multiple OCSP URIs in one certificate (HTTP level high-availability is used), I would not consider necessary to support it for local OCSP mirrors.